tools/clang/test/HLSLFileCheck/hlsl/control_flow/attributes/unroll/lib_2d_array_call_export.hlsl

```diff
@@ -0,0 +1,53 @@
+// RUN: %dxc -T lib_6_x %s | FileCheck %s
+
+// Ensure 2d array is supported.
+
+// Ensure BreakUpArrayAllocas doesn't misfire and ignore the function call
+// users, causing everything but the calls to be moved to separate scalar
+// arrays, then all that code, being dead, getting deleted, leaving just the
+// passing of uninitialized arrays to the function calls.
+
+// CHECK-DAG: %[[h0:.*]] = load %dx.types.Handle, %dx.types.Handle* @{{.*}}buf0
+// CHECK-DAG: %[[h1:.*]] = load %dx.types.Handle, %dx.types.Handle* @{{.*}}buf1
+// CHECK-DAG: %[[h2:.*]] = load %dx.types.Handle, %dx.types.Handle* @{{.*}}buf2
+// CHECK-DAG: %[[h3:.*]] = load %dx.types.Handle, %dx.types.Handle* @{{.*}}buf3
+// CHECK-DAG: %[[gep0:.*]] = getelementptr inbounds [2 x [2 x %dx.types.Handle]], [2 x [2 x %dx.types.Handle]]* %buffers, i32 0, i32 0, i32 0
```

Comment:
Is it okay to just hardcode the name of the value to be %buffer?

Reply:
Names will/may be stripped in release build, which reminds me this uses `%buffers` and that could be stripped (I think), so I should fix that at least.